src/firefly/java/edu/caltech/ipac/firefly/server/query/mos/QueryMOS.java

```diff
@@ -187,11 +187,17 @@ private URL createURL(MOSRequest req, boolean forGator) throws EndUserException,
         return new URL(url);
     }
 
+    //Add check for empty date string and allow null string for obsDateParam
     private static String getObsDate(String obsDateParam) {
         try {
-            DateUtils.parseDate(obsDateParam, new String[]{"yyyy-MM-dd", "yyyy-MM-dd hh:mm:ss", "yyyy-MM-dd'T'hh:mm:ssX"});     // check for valid dates
-            return obsDateParam.replaceAll("[^0-9:-]", " ").trim();   // convert to format expected by MOST
-        } catch (ParseException e) {
+                if (StringUtils.isEmpty(obsDateParam)) {
+                    return null;
+            } else {
+                DateUtils.parseDate(obsDateParam, new String[]{"yyyy-MM-dd", "yyyy-MM-dd hh:mm:ss", "yyyy-MM-dd'T'hh:mm:ssX"});     // check for valid dates
+                return obsDateParam.replaceAll("[^0-9:-]", " ").trim();   // convert to format expected by MOST
+            }
+        }
+         catch (ParseException e) {
```

Comment:
This would be fine, but since we're already using ParseException as a validation mechanism, why not add IllegalArgumentException to the catch clause, or better yet, just catch Exception instead.

https://commons.apache.org/proper/commons-lang/javadocs/api-2.2/